lib/platform/linux-i2c.c

```diff
@@ -47,17 +47,33 @@ struct switchtec_i2c {
 	struct switchtec_dev dev;
 	int fd;
 	int i2c_addr;
+	uint8_t tag;
 };
 
 #define CMD_GET_CAP  0xE0
+#define CMD_GAS_WRITE  0xEA
+#define CMD_GET_WRITE_STATUS  0xE2
+#define CMD_GAS_WRITE_WITH_STATUS  0xE8
+
 #define MAX_RETRY_COUNT  100
 #define PEC_BYTE_COUNT  1
 #define GAS_TWI_MODE_MASK  0x80
+#define GAS_TWI_MRPC_ERR  0x20
 
 #define to_switchtec_i2c(d)  \
 	((struct switchtec_i2c *) \
 	 ((char *)d - offsetof(struct switchtec_i2c, dev)))
 
+static uint8_t get_tag(struct switchtec_dev *dev)
+{
```

Comment:
Probably better for this function to take the switchtec_i2c struct and let the caller do the conversion.